tiatoolbox/models/dataset/dataset_abc.py

```diff
@@ -143,11 +145,14 @@ def preproc_func(self, func):
         elif callable(func):
             self._preproc = func
         else:
-            raise ValueError(f"{func} is not callable!")
+            msg = f"{func} is not callable!"
+            raise ValueError(msg)
 
-    def __len__(self):
+    def __len__(self) -> int:
+        """Returns the length of the instance attributes."""
```

Comment:
```suggestion
        """Return the length of the instance attributes."""
```

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/625/commits/2db02db7a6949e400ee54fe03a5dfa0021412f44